databrowser/src/pages/datasets/overview/OverviewListPage.vue

```diff
@@ -59,21 +174,445 @@ SPDX-License-Identifier: AGPL-3.0-or-later
 import OverviewCardItem from './OverviewCardItem.vue';
 import PartnersAndContributors from '../../../components/partners/PartnersAndContributors.vue';
 import CardDivider from '../../../components/card/CardDivider.vue';
 import PageGridContent from '../../../components/content/PageGridContent.vue';
 import { useI18n } from 'vue-i18n';
 import { useMetaDataDatasets, useOtherDatasets } from './useDatasets';
+import ToggleCustom from '../../../components/toggle/ToggleCustom.vue';
+import InputCustom from '../../../components/input/InputCustom.vue';
+import Accordion from '../../../components/accordion/Accordion.vue';
+import CheckboxCustom from '../../../components/checkbox/CheckboxCustom.vue';
+import IconFilter from '../../../components/svg/IconFilter.vue';
+import IconClose from '../../../components/svg/IconClose.vue';
+import InfoPopover from '../../../components/popover/InfoPopover.vue';
+import PopoverCustomPanel from '../../../components/popover/PopoverCustomPanel.vue';
+import PopoverContent from '../../../components/popover/PopoverContent.vue';
+import { TourismMetaData } from '../../../domain/metaDataConfig/tourism/types';
+import { DatasetConfig } from '../../../domain/datasetConfig/types';
+import ResetAllFilters from '../../../domain/datasets/tableView/filter/ResetAllFilters.vue';
 
-const { t } = useI18n();
+type TourismMetaDataIndexes =
+  | 'dataSpace'
+  | 'categories'
+  | 'tags'
+  | 'datasetConfigurations'
+  | 'access'
+  | 'sources'
+  | 'dataProviders'
+  | 'singleDataset';
+
+interface Filters {
+  searchVal: string;
+  applied: Record<string, FilterObject>;
+}
+
+interface FilterObject {
+  key: string;
+  value?: string;
+}
+
+interface FilterSelectItem {
+  key: string;
+  value: string;
+}
+
+interface DynamicFilter {
+  id: string;
+  name: string;
+  data: FilterSelectItem[];
+}
+
+const defaultFilters = {
+  searchVal: '',
+  applied: {} as Record<string, FilterObject>,
+};
+const _inputModels = ref<Record<string, boolean>>({
+  no_metadata: false,
+  deprecated: false,
+});
+
+const isFiltersModalVisible = ref(false);
+const filters = ref(structuredClone(defaultFilters) as Filters);
+
+// Methods
+const setInputModelKeyValue = (
+  key: TourismMetaDataIndexes,
+  propName: string
+) => {
+  // NOTE: defining the object key here it's necessary to properly reset the object programmatically with the reset button. Using the filters array to handle component's v-model will lead into too much code complexity.
+  _inputModels.value[getInputModelId(key as TourismMetaDataIndexes, propName)] =
+    false;
+};
+
+const mapDatasetsKeyArrayToFilterItems = (
+  datasets: TourismMetaData[],
+  key: TourismMetaDataIndexes
+) => {
+  const addedProps: Record<string, boolean> = {};
+  let finalList = [] as FilterSelectItem[];
+
+  datasets.forEach((dataset) => {
+    if (dataset[key] !== undefined) {
+      (dataset[key]! as string[]).forEach((propName) => {
```

Comment:
Exclamation mark can be removed, as you have checked for existence beforehand.

Minor point: you just check for `undefined` in the line above, what about `null` values? You can check both for `undefined` and `null` at once using the `!= null` check like this:

```js
if (dataset[key] != null) {
```

That's 95% of the time the kind of check that you want to do

Reply:
I thought strict equality was required. Refactored. 👍